Cython/Compiler/Parsing.py

```diff
@@ -2141,6 +2141,10 @@ def p_with_items(s, is_async=False):
             s.next()
             items = p_with_items_list(s, is_async)
             s.expect(")")
+            if s.sy != ":":
+                # Fail - the message doesn't matter because we'll try the
+                # non-bracket version so it'll never be shown
+                s.error("")
```

Comment:
Isn't this mostly just the following?
```suggestion
            # I the command isn't complete: fail, then try the non-bracket version.
            s.expect(":")
```

Reply:
I don't think it quite is. `expect` consumes the character I think, while I want to leave that to `p_suite`